src/Operation/AbstractOperation.php

```diff
@@ -87,8 +91,11 @@ protected function initRequest(
         $request->setResponseTimeout($this->getResponseTimeout());
         $request->setConnectTimeout($this->getConnectTimeout());
         $request->setContentType($this->getContentType());
-        $request->setAuthBasic('', $token->getValue());
+        if (!is_null($token) && $isTokenNeeded) {
+            $request->setAuthBasic('', $token->getValue());
+        }
         $request->setPath($this->getPath());
+        $request->setResultHandler($this->getResultHandler());
```

Comment:
This pattern isn't logical. You're setting the exception handler in the base-class and then you leverage it in the parent class? If you're sitting here, shouldn't you be using it here too?